tests/end_to_end_test/server_metrics_test.go

```diff
@@ -0,0 +1,279 @@
+package endtoend_test
+
+import (
+	"fmt"
+	"io"
+	"net/http"
+	"testing"
+
+	"github.com/stretchr/testify/require"
+
+	"github.com/kopia/kopia/internal/apiclient"
+	"github.com/kopia/kopia/internal/serverapi"
+	"github.com/kopia/kopia/internal/testlogging"
+	"github.com/kopia/kopia/internal/testutil"
+	"github.com/kopia/kopia/tests/testenv"
+)
+
+func TestServerMetrics(t *testing.T) {
+	ctx := testlogging.Context(t)
+
+	runner := testenv.NewInProcRunner(t)
+	e := testenv.NewCLITest(t, testenv.RepoFormatNotImportant, runner)
+
+	defer e.RunAndExpectSuccess(t, "repo", "disconnect")
+
+	e.RunAndExpectSuccess(t, "repo", "create", "filesystem", "--path", e.RepoDir, "--override-hostname=fake-hostname", "--override-username=fake-username")
+
+	e.RunAndExpectSuccess(t, "snapshot", "create", sharedTestDataDir1)
+	e.RunAndExpectSuccess(t, "snapshot", "create", sharedTestDataDir2)
+
+	var sp testutil.ServerParameters
+
+	wait, _ := e.RunAndProcessStderr(t, sp.ProcessOutput,
+		"server", "start",
+		"--ui",
+		"--address=localhost:0",
+		"--random-password",
+		"--random-server-control-password",
+		"--tls-generate-cert",
+		"--tls-generate-rsa-key-size=2048",      // use shorter key size to speed up generation
+		"--metrics-listen-addr=localhost:10000", // enable metrics
```

Comment:
It's usually better to use port zero in tests to avoid any conflict with any process holding the particular port open.

Reply:
I do agree, and indeed I did parsing of the random port,, but setting port to 0 it will be printed as 0 in server startup. I did not investigate further and set to 10000.

Reply:
you need to print the actual port number it binds to same as we do elsewhere.

Reply:
Done